### PR TITLE
Auto detect mutators

### PR DESCRIPTION
**Pull request by YuxiangChai**

### CODE REVIEW

**Comment by pkmoore** [comment]
I get one failing test on this branch:
```
======================================================================
ERROR: Failure: ValueError (Attempted relative import beyond toplevel package)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/preston/rrapper/crashsim/local/lib/python2.7/site-packages/nose-1.3.7-py2.7.egg/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/preston/rrapper/crashsim/local/lib/python2.7/site-packages/nose-1.3.7-py2.7.egg/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/preston/rrapper/crashsim/local/lib/python2.7/site-packages/nose-1.3.7-py2.7.egg/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/preston/rrapper/test_inject.py", line 14, in <module>
    from inject import exit_with_status
  File "/home/preston/rrapper/src/inject.py", line 51, in <module>
    from Mutators import *
  File "/home/preston/rrapper/src/Mutators/test_UnusualFiletype.py", line 11, in <module>
    from ..consts import DEFAULT_CONFIG_PATH
ValueError: Attempted relative import beyond toplevel package

----------------------------------------------------------------------
```

File: src/Mutators/.coverage [hunk 1]
```diff
@@ -0,0 +1 @@
...01,303,330,335,338,340,351,354,356,367,370,372,383,384,472,500],"/home/chai/crashsim/lib/python2.7/site-packages/sysDef/SyscallParameter.py":[1,50,52,220,169],"/home/chai/rrapper/src/mutator/mutator.py":[1,3,4,38,41,45,77,81,51,21,22,68,26,53],"/home/chai/crashsim/lib/python2.7/site-packages/sysDef/SyscallManual.py":[65,39,41,42,43,108,45,110,109,141,49,114,51,436,105,408,111,61,62,426],"/home/chai/crashsim/lib/python2.7/site-packages/sysDef/Definition.py":[24,1,3,22,95],"/home/chai/crashsim/lib/python2.7/site-packages/posix_omni_parser/parsers/StraceParser.py":[1028,1031,523,528,88,535,536,27,29,542,31,32,35,38,563,568,570,573,574,693,577,67,581,584,588,592,593,594,596,598,600,103,104,105,621,119,122,151,152,154,159,675,677,680,681,687,178,692,181,698,699,722,742,894,811,776,778,779,783,784,785,786,789,792,795,810,903,814,815,822,823,315,316,317,318,319,320,321,322,837,539,328,329,331,336,337,852,341,346,859,862,351,865,354,355,361,366,371,372,887,891,380,381,382,383,384,897,901,391,906,398,399,404,413,416,419,65,423,937,428,429,431,952,968,971,163,468,992,848,422,1008],"/home/chai/rrapper/src/mutator/test_unusual.py":[1,2,3,4,6,7,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,26]}}
```
Comment (pkmoore) [inline src/Mutators/.coverage]: This looks like a config file for your editor or something.  Please remove it from this commit

File: src/Mutators/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import glob
```
Comment (pkmoore) [inline src/Mutators/__init__.py]: Is this what correctly gathers the mutators so they can be imported with * later on?

File: src/rrtest.py [hunk 3]
```diff
@@ -32,15 +32,9 @@
 import shutil
 import errno
 import ConfigParser
-
+
```
Comment (pkmoore) [inline src/rrtest.py]: Clean up whitespace here

File: src/rrtest.py [hunk 4]
```diff
@@ -206,6 +200,7 @@ def configure_test(name, mutator, verbosity, trace_line=0, sniplen=5):
     if mutator:
       #config.set("request_handling_process", "mutator", args.mutator)
       # use the mutator to identify the line we are interested in
+      mutator = mutator[:-9] + '.' + mutator
```
Comment (pkmoore) [inline src/rrtest.py]: What's going on here?  Can we add a comment to describe why we want to do this?

**Review by pkmoore — CHANGES_REQUESTED** [review]
I'm good with this strategy for automatically importing all mutators.  I like it because it uses default python machinery 👍.

My only concern is because we're messing with import stuff we need to make sure we haven't broken something.  Can you run some recordings and replays to make sure all our imports work correctly? Once that's done, and my other review comments are addressed, we're ready to merge this.

Thanks!

File: src/Mutators/__init__.py (hunk 2 above)
Comment (YuxiangChai) [reply src/Mutators/__init__.py]: This is a Unix style pathname pattern expansion module in python, to make a path look like '/.../.../*.py'

**Comment by pkmoore** [comment]
Looks good.  Thanks!

**Pull request merged**